src/shared/tpm2-util.c

```diff
@@ -476,6 +476,16 @@ void tpm2_pcr_mask_to_selection(uint32_t mask, uint16_t bank, TPML_PCR_SELECTION
         };
 }
 
+#define tpm2_log_debug_digest(digest, fmt, ...)                         \
+        ({                                                              \
+                if (DEBUG_LOGGING) {                                    \
+                        _cleanup_free_ char *_h_ = hexmem((digest)->buffer, (digest)->size); \
```

Comment:
we usually want our macros side-effect free, hence maybe assign `digest` to a temporary variable first, before using it, so that it never gets evaluated twice.

Reply:
ack